### PR TITLE
Fix bug with inline comment

### LINKED ISSUE #167: Bug with " symbol in inline comment
Library version: 0.3.0

Can be reproduced with a test:
```
    @Test
    fun symbolsAfterComment() {
        val keyValue = "test_key = \"test_value\"  # \" some comment".splitKeyValue(0)
        assertEquals("test_value", TomlKeyValuePrimitive(keyValue, 0).value.content)
    }
```
This is a valid TOML format, but I get an exception:

```
Line 0: According to the TOML specification string values (even Enums) should be wrapped (start and end) with quotes (""), but the following value was not: <"test_value"  # " some comment>. Please note that multiline strings are not yet supported.
com.akuleshov7.ktoml.exceptions.ParseException: Line 0: According to the TOML specification string values (even Enums) should be wrapped (start and end) with quotes (""), but the following value was not: <"test_value"  # " some comment>. Please note that multiline strings are not yet supported.
```

Where the problem is: file `TomlKeyValue`, `splitKeyValue` function.

### PR DESCRIPTION
**Pull request by BOOMeranGG**
Fix #167

### CODE REVIEW

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/StringUtils.kt [hunk 1]
```diff
@@ -167,6 +173,33 @@ private fun String.validateSymbols(lineNo: Int) {
     }
 }
 
+private fun String.getCommentStartIndex(): Int {
+    // temporary replace \" and \' to hide non-significant quotes
+    // we cannot remove them because we need to know the original index
+    val chars = this
+        .replace("\\\"", "__")
+        .replace("\\\'", "__")
```
Comment (NightEule5) [inline ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/StringUtils.kt:181]: The TOML spec allows backslashes to be interpreted literally in single-quoted strings, we just have a config option to enable single quote escaping. Would this work with, say, `a = 'C:\some\path\'`?

Comment (BOOMeranGG) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/StringUtils.kt:181]: Good point! It doesn't work for this test

Comment (BOOMeranGG) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/StringUtils.kt]: Fixed. You can see additional tests in StringUtilsTest, now they work

**Comment by BOOMeranGG** [comment]
@akuleshov7 Previous time one check failed. Is something wrong?

**Comment by akuleshov7** [comment]
> @akuleshov7 Previous time one check failed. Is something wrong?
approved run, let me check. Have you run `./gradlew diktatFixAll` or `./gradlew detekt` tasks?

**Review by akuleshov7 — APPROVED** [review]

**Comment by BOOMeranGG** [comment]
> Have you run ./gradlew diktatFixAll or ./gradlew detekt tasks?
Yes, I have. It's all right. Let's merge?

**Comment by akuleshov7** [comment]
> Have you run ./gradlew diktatFixAll or ./gradlew detekt tasks?
> Yes, I have. It's all right. Let's merge?

Great, thank you so much for this fix!

**Pull request merged**